keps/sig-network/20190423-service-lb-finalizer.md

```diff
@@ -0,0 +1,156 @@
+---
+title: Finalizer Protection for Service LoadBalancers
+authors:
+  - "@MrHohn"
+owning-sig: sig-network
+participating-sigs:
+  - sig-cloud-provider
+reviewers:
+  - "@andrewsykim"
+  - "@bowei"
+  - "@jhorwit2"
+  - "@jiatongw"
+approvers:
+  - "@andrewsykim"
+  - "@bowei"
+  - "@thockin"
+editor: TBD
+creation-date: 2019-04-23
+last-updated: 2019-04-23
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Finalizer Protection for Service LoadBalancers
+
+## Table of Contents
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Release Signoff Checklist](#release-signoff-checklist)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+  - [Proposal](#proposal)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+We will be adding finalizer protection to ensure the Service resource is not
+fully deleted until the correlating load balancing resources are also deleted.
```

Comment:
```suggestion
fully deleted until the correlating load balancing resources are deleted.
```
this seems better?

Reply:
Thanks, I will apply these and some suggestions above at the next push.

Reply:
Fixed.

Reply:
👍